### ISSUE TITLE
Implement spread prediction regression model

### ISSUE THREAD

**Issue by muk2** [Kame4201/beat-books-model#20]
### Why this matters
**Non-technical:** Sports betting isn't just about picking winners — the spread (point difference) is how most bets work. Right now, our system can predict *who* wins but not *by how much*. Spread prediction unlocks the most common and potentially profitable betting market: against-the-spread (ATS) bets.

**Technical:** There's a `TODO` in `src/backtesting/backtester.py:305`:
```python
predicted_spread=0.0,  # TODO: implement spread prediction
```

The `SpreadModel` class exists in PR #6 (`src/models/spread_model.py`) but isn't integrated into the backtesting pipeline. The backtester currently only uses `predict_proba()` for win/loss probability and hardcodes spread to 0.0.

### Implementation Plan

1. **Extend the `Model` Protocol** in `backtester.py`:
   ```python
   class Model(Protocol):
       def fit(self, X: pd.DataFrame, y: pd.Series) -> None: ...
       def predict_proba(self, X: pd.DataFrame) -> np.ndarray: ...
       def predict_spread(self, X: pd.DataFrame) -> np.ndarray: ...  # NEW
   ```

2. **Integrate spread predictions into backtesting**:
   - Call `model.predict_spread()` alongside `predict_proba()`
   - Use predicted spread to make ATS betting decisions
   - Compare predicted spread vs market spread for edge detection
   - Track ATS-specific metrics (cover rate, ATS ROI)

3. **Add spread-specific betting logic**:
   - Bet ATS when predicted spread differs significantly from market spread
   - Use Kelly Criterion on spread bets (different edge calculation)
   - Track separate bankroll or combined bankroll for moneyline + ATS

4. **Add spread-specific metrics**:
   - Mean Absolute Error (MAE) of spread predictions
   - Cover rate (how often our predicted side covers)
   - ATS ROI separate from moneyline ROI

### Acceptance Criteria
- [ ] Backtester supports both win/loss and spread predictions
- [ ] Spread predictions integrated into betting simulation
- [ ] Spread-specific metrics (MAE, cover rate, ATS ROI) tracked
- [ ] Unit tests for spread prediction flow
- [ ] TODO in backtester.py resolved

### Dependencies
- Requires PR #6 (SpreadModel) and PR #7 (Backtesting) to be merged

**Comment by Kame4201** [comment]
**Closing as done.**

**Evidence:**
- `src/models/spread_model.py` fully implemented on Dev with 4 model variants (LinearRegression, Ridge, XGBoost, LightGBM)
- PR #39 (`feature/issue-20-spread-backtesting`) integrates spread predictions into backtesting pipeline — open, all CI checks pass, ready for review
- Tests: `tests/test_models/test_spread_model.py`, `tests/test_backtesting/test_spread_integration.py`
- 183 tests pass on PR #39 branch

Note: PR #39 still needs merge to complete the backtesting integration, but the spread model itself is fully implemented.
